src/common/snippets/src/lowered/pass/insert_specific_iterations.cpp

```diff
@@ -17,6 +19,37 @@ namespace snippets {
 namespace lowered {
 namespace pass {
 
+namespace {
+void connect_cloned_body_with_buffers_outside(LinearIR::constExprIt cur_begin, LinearIR::constExprIt cur_end,
+                                              LinearIR::constExprIt res_begin, LinearIR::constExprIt res_end,
+                                              LinearIR& linear_ir) {
+    for (LinearIR::constExprIt result_it = res_begin, original_it = cur_begin; result_it != res_end; ++result_it, ++original_it) {
+        const auto result_expr = *result_it;
+        const auto original_expr = *original_it;
```

Comment:
Minor: what's wrong with `auto` 🙂 + references?
```suggestion
    for (auto result_it = res_begin, original_it = cur_begin; result_it != res_end; ++result_it, ++original_it) {
        const auto& result_expr = *result_it;
        const auto& original_expr = *original_it;
```

Reply:
Fixed in https://github.com/openvinotoolkit/openvino/pull/24646/commits/a9fb56e5b7e051ecaa58b2f7f73d84ba0d698d28